.circleci/config.yml

```diff
@@ -71,212 +232,564 @@ jobs:
+                    SDK_VERSION=$(mvn -q -Dexec.executable="echo" -Dexec.args='${project.version}' --non-recursive exec:exec)
+                    
+                    git commit -am "updated minor version to v${SDK_VERSION}-SNAPSHOT [skip ci]"
+                    git checkout -b minor-snapshot-update
+                    git merge --strategy=ours develop
+                    git checkout develop
+                    git merge minor-snapshot-update
+                    git tag -a v${SDK_VERSION}-SNAPSHOT -m "v${SDK_VERSION}-SNAPSHOT release"
+                    git push --set-upstream origin develop v${SDK_VERSION}-SNAPSHOT
+
                     exit 0
                     ;;
-                ( !(!(*"patch"*)|!(*"release"*)) )
-                    echo "deploy patch release"
+                *"[patch]"*)
+
+                    echo "bump patch release version"
                     mvn build-helper:parse-version versions:set -DnewVersion=\${parsedVersion.majorVersion}.\${parsedVersion.minorVersion}.\${parsedVersion.nextIncrementalVersion} versions:commit
-                    echo "TODO - deploy to jitpack command goes here"
                     SDK_VERSION=$(mvn -q -Dexec.executable="echo" -Dexec.args='${project.version}' --non-recursive exec:exec)
-                    git commit -am "released v${SDK_VERSION}-SNAPSHOT [skip ci]"
-                    git tag -a v${SDK_VERSION} -m "release v${SDK_VERSION}-SNAPSHOT"
-                    git push --set-upstream origin develop v${SDK_VERSION}
+                    
+                    git commit -am "updated patch version to v${SDK_VERSION}-SNAPSHOT [skip ci]"
+                    git checkout -b patch-snapshot-update
+                    git merge --strategy=ours develop
+                    git checkout develop
+                    git merge patch-snapshot-update
+                    git tag -a v${SDK_VERSION}-SNAPSHOT -m "v${SDK_VERSION}-SNAPSHOT release"
+                    git push --set-upstream origin develop v${SDK_VERSION}-SNAPSHOT
+
                     exit 0
                     ;;
                 *)
-                    echo "do nothing"
+                    echo "This was just a simple merge - no versions updated"
                     exit 0
                     ;;
             esac
+
+
+  publish-release-version:
+    steps:
+      - run: sudo apt install maven -y
+      - run:
+          name: Create and push release version tag
+          command: |
+            git branch
+            SDK_VERSION=$(mvn -q -Dexec.executable="echo" -Dexec.args='${project.version}' --non-recursive exec:exec)
+            echo $SDK_VERSION
+            git fetch
+            git checkout master
+            git reset --hard release
+            git push origin :release
+            git tag -a v${SDK_VERSION} -m "v${SDK_VERSION} (stable) release"
+            git push --force --set-upstream origin master v${SDK_VERSION}
```

Comment:
My advice is still that you should merge the release into master instead of force-pushing to it. If the merge fails because of conflicts then you will at least know about it and be able to track the problem. And the merge should never fail, since no one should directly commit to master - but if it does, you will at least know about it. When force pushing automatically, you might just silently overwrite something and cause confusion later.